### PR TITLE
fix: fw-618 redpacket post keep recreating when the post gets focus

### PR DESCRIPTION
**Pull request by swkatmask**
https://mask.atlassian.net/browse/fw-618

### CODE REVIEW

File: src/components/Compose/ComposeImage.tsx [hunk 1]
```diff
@@ -23,7 +23,7 @@ export const ComposeImage = memo(function ComposeImage({ index, size, image, rea
                 size <= 2 ? ' h-72' : size === 3 && index === 2 ? ' h-72' : ' h-[138px]',
                 size === 1 ? ' col-span-2' : '',
                 size === 3 && index === 1 ? ' col-start-1' : '',
-                size === 3 && index === 2 ? ' absolute right-3 top-3 w-[251px]' : ' relative',
+                size === 3 && index === 2 ? ' absolute right-0 top-3 w-[calc(50%-0.5rem)]' : ' relative',
```
Comment (guanbinrui) [inline src/components/Compose/ComposeImage.tsx:26]: Avoid using `rem`.

File: src/modals/ComposeModal.tsx [hunk 2]
```diff
@@ -177,24 +179,17 @@ export const ComposeModalUI = forwardRef<SingletonModalRefCreator<ComposeModalPr
                     }),
                 ].join('\n');
 
-                updateChars([
```
Comment (guanbinrui) [inline src/modals/ComposeModal.tsx:180]: Why delete this?

File: src/modals/ComposeModal.tsx [hunk 3]
```diff
@@ -141,8 +141,10 @@ export const ComposeModalUI = forwardRef<SingletonModalRefCreator<ComposeModalPr
             }
         }, [posts, dispatch]);
 
+        // To avoid recreating post content for Redpacket
+        const encryptedMapRef = useRef<Map<string, boolean>>(new Map());
```
Comment (guanbinrui) [inline src/modals/ComposeModal.tsx:145]: Use compose store stead.

File: src/components/Compose/ComposeImage.tsx (hunk 1 above)
Comment (swkatmask) [reply src/components/Compose/ComposeImage.tsx:26]: `0.5rem` is exactly width of the gap.

File: src/modals/ComposeModal.tsx (hunk 2 above)
Comment (swkatmask) [reply src/modals/ComposeModal.tsx:180]: The added text will be overwritten by the following line.

File: src/modals/ComposeModal.tsx (hunk 3 above)
Comment (swkatmask) [reply src/modals/ComposeModal.tsx:145]: It might not worthy to put it into the store.

Comment (guanbinrui) [reply src/modals/ComposeModal.tsx:145]: I think the current store should able to handle this.

File: src/components/Compose/ComposeImage.tsx (hunk 1 above)
Comment (guanbinrui) [reply src/components/Compose/ComposeImage.tsx:26]: Then please use gap.

File: src/modals/ComposeModal.tsx (hunk 2 above)
Comment (guanbinrui) [reply src/modals/ComposeModal.tsx:180]: This is required to add a hidden label for rp posts.

File: src/modals/ComposeModal.tsx [hunk 4]
```diff
@@ -185,16 +188,19 @@ export const ComposeModalUI = forwardRef<SingletonModalRefCreator<ComposeModalPr
...
                     fullMessage,
                 ]);
+
                 setEditorContent(fullMessage);
 
                 addImage({
                     file: new File([secretImage], 'image.png', { type: 'image/png' }),
                 });
+
+                trackEncryptedRedPacket(id);
```
Comment (guanbinrui) [inline src/modals/ComposeModal.tsx:198]: Delete `typedMessage` of the post,  after attaching chars and images.

**Pull request merged**